EveryDream_Tools_4_colab.ipynb

```diff
@@ -0,0 +1,561 @@
+        "\n",
+        " * The COCO dataset consists of more than 330,000 images with five captions each, and is known for its high-quality captions and diversity of image types. \n",
+        "\n",
+        "\n",
+        "* \"T5-FLAN\" is similar to \"Blip\" in that it is trained on a large and diverse dataset of image captions. However, it uses a different architecture called the \"FLAN\" (Feature-wise Linear Attention) transformer. The FLAN transformer is designed to better capture long-range dependencies and improve the modeling of feature interactions in the encoder, which can lead to better performance on certain tasks.\n",
+        "\n",
+        "* \"Git Captioner\" is a model that generates image captions using a combination of computer vision and natural language processing techniques. It leverages pre-trained models for object detection and recognition, and then generates captions using a transformer-based language model.\n",
+        "\n",
+        "When choosing among these models, consider your specific task or dataset needs. If you require a model that can perform well across a wide range of image captioning tasks and datasets, \"Blip\" or \"T5-FLAN\" may be a good choice. If you require strong performance specifically on the COCO dataset, consider fine-tuning \"Blip-COCO\". If you prefer a model that leverages object detection and recognition in addition to language modeling, consider \"Git Captioner\".\n",
+        "\n",
+        "In addition to image captioning, \"T5-FLAN\" and \"Git Captioner\" have also been applied to tasks such as image generation and visual question answering.\n",
+        "\n"
+      ],
+      "metadata": {
+        "id": "NPWIluxMhAm8"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "#@title Blip2 uses alot of ram, this cell will clear our ram usage\n",
+        "#@markdown Run this cell before captioning and if your model crashes while loading.\n",
+        "import gc\n",
+        "gc.collect()"
+      ],
+      "metadata": {
+        "cellView": "form",
+        "id": "23gKI8QGJTyM"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "4TAICahl-RPn",
+        "cellView": "form"
+      },
+      "outputs": [],
+      "source": [
+        "#@title Dream Captioner\n",
+        "input_folder = \"/content/EveryDream/input\" #@param {type:\"string\"}\n",
+        "#@markdown * The location of images to be captoned\n",
+        "model_name = \"Salesforce/blip2-opt-2.7b-coco\" #@param [\"microsoft/git-large-r-textcaps\", \"microsoft/git-large-textcaps\", \"microsoft/git-base-textcaps\", \"Salesforce/blip2-opt-2.7b\", \"Salesforce/blip2-opt-2.7b-coco\", \"Salesforce/blip2-flan-t5-xl\"]\n",
```

Comment:
I think the default should be something that will run on Colab Free tier.  Git-large-textcaps seems to work, probably best one, or git-large-t-textcaps.